src/components/document/sort-work-document-area.tsx

```diff
@@ -0,0 +1,91 @@
+import React from "react";
+import classNames from "classnames";
+import { useAppConfig, useProblemStore,
+  usePersistentUIStore, useUserStore, useClassStore, useUIStore, useStores } from "../../hooks/use-stores";
+import { DocumentModelType } from "../../models/document/document";
+import { EditableDocumentContent } from "./editable-document-content";
+import { getDocumentDisplayTitle } from "../../models/document/document-utils";
+import EditIcon from "../../clue/assets/icons/edit-right-icon.svg";
+import { observer } from "mobx-react";
+import { tabName } from "./sort-work-view";
+
+
+export const SortWorkDocumentArea: React.FC = observer(function SortWorkDocumentArea() {
+  const store = useStores();
+  const ui = useUIStore();
+  const persistentUI = usePersistentUIStore();
+  const user = useUserStore();
+  const classStore = useClassStore();
+  const problemStore = useProblemStore();
+  const appConfigStore = useAppConfig();
+
+  const navTabSpec = appConfigStore.navTabs.getNavTabSpec(tabName);
+  const tabState = navTabSpec && persistentUI.tabs.get(navTabSpec?.tab);
+  const openDocumentKey = tabState?.openDocuments.get(tabName) || "";
```

Comment:
Given that SortWorkView is basically already figuring out the openDocumentKey, you might as well pass it in as a property to SortWorkDocumentArea. This way SortWorkDocumentArea doesn't have to use `appConfigStore.navTabs`, or `persistentUI`.